src/aiortc/contrib/media.py

```diff
@@ -679,69 +804,114 @@ async def __run_track(self, track, context):
+                if track.kind == "keypoints":
+                    # keypoint stream
+                    received_keypoints = frame.data
+                    asyncio.run_coroutine_threadsafe(self.__keypoints_queue.put(received_keypoints), loop)
+                    frame_index = received_keypoints['frame_index']
+
+                    if save_keypoints_to_file:
+                        keypoints_file = open(self.__keypoints_file_name, "a")
+                        keypoints_file.write(str(received_keypoints))
+                        keypoints_file.write("\n")
+                        keypoints_file.close()
+
+                elif track.kind == "lr_video":
+                    lr_frame, frame_index = destamp_frame(frame)
+                    lr_frame_array = lr_frame.to_rgb().to_ndarray()
+                    asyncio.run_coroutine_threadsafe(self.__lr_video_queue.put((lr_frame_array, frame_index)), loop)
+                    if save_lr_video_npy and self.__save_dir is not None:
+                        np.save(os.path.join(self.__save_dir,
+                                'receiver_lr_frame_%05d.npy' % frame_index),
+                                lr_frame_array)
+ 
+                self.__log_debug("%s for frame %s received at time %s",
+                                track.kind, str(frame_index), datetime.datetime.now())
 
                 if self.__enable_prediction:
-                    self.__setsize(track)
                     try:
-                        received_keypoints = await self.__keypoints_queue.get()
-                        frame_index = received_keypoints['frame_index']
-
-                        if frame_index % self.__reference_update_freq == 0:
-                            source_frame_array, source_keypoints, video_frame_index = await self.__video_queue.get()
-                            
-                            time_before_update = time.perf_counter()
-                            model.update_source(video_frame_index, source_frame_array, source_keypoints)
-                            time_after_update = time.perf_counter()
-                            self.__log_debug("Time to update source frame %s in receiver" \
-                                    " when receiving keypoint %s: %s",
-                                    video_frame_index, frame_index, str(time_after_update - time_before_update))
+                        if track.kind == "keypoints":
+                            received_keypoints = await self.__keypoints_queue.get()
+                            frame_index = received_keypoints['frame_index']
+                        elif track.kind == "lr_video":
+                            lr_frame_array, frame_index = await self.__lr_video_queue.get()
+                        if self.__display_option == "synthatic":
+                            if frame_index % self.__reference_update_freq == 0 and \
+                                    generator_type not in ['vpx', 'bicubic']:
```

Comment:
if "generator" is in vpx, you should never even get here right?

Reply:
correct, I just copied and pasted the `generator_type not in ['vpx', 'bicubic']` condition there